x-pack/plugins/snapshot_restore/public/application/components/feature_states_form_field/feature_states_form_field.tsx

```diff
@@ -0,0 +1,132 @@
+  EuiSwitch,
+  EuiIconTip,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+
+import { useServices } from '../../app_context';
+import { FEATURE_STATES_NONE_OPTION } from '../../../../common/constants';
+import { SlmPolicyPayload, RestoreSettings } from '../../../../common/types';
+
+export type FeaturesOption = EuiComboBoxOptionOption<string>;
+
+interface Props {
+  featuresOptions: FeaturesOption[];
+  selectedOptions: FeaturesOption[];
+  setSelectedOptions: (features: FeaturesOption[]) => void;
+  onUpdateFormSettings: (
+    arg: Partial<SlmPolicyPayload['config']> & Partial<RestoreSettings>
+  ) => void;
+  hasNoneOptionSelected: boolean;
+  isLoadingFeatures?: boolean;
+}
+
+export const FeatureStatesFormField: FunctionComponent<Props> = ({
+  isLoadingFeatures = false,
+  featuresOptions,
+  selectedOptions,
+  setSelectedOptions,
+  onUpdateFormSettings,
+  hasNoneOptionSelected,
+}) => {
+  const { i18n } = useServices();
+
+  const onChange = (selected: FeaturesOption[]) => {
+    setSelectedOptions(selected);
+    onUpdateFormSettings({
+      featureStates: selected.map((option) => option.label),
+    });
+  };
+
+  const onIncludeNoneSwitchChange = () => {
+    if (!hasNoneOptionSelected) {
+      setSelectedOptions([{ label: FEATURE_STATES_NONE_OPTION }]);
+      onUpdateFormSettings({ featureStates: [FEATURE_STATES_NONE_OPTION] });
+    } else {
+      setSelectedOptions([]);
+      onUpdateFormSettings({ featureStates: [] });
+    }
+  };
+
+  return (
+    <EuiFormRow
+      label={
+        <>
+          <FormattedMessage
+            id="xpack.snapshotRestore.featureStatesFormField.formRowLabel"
+            defaultMessage="Include feature states from"
+          />{' '}
+          <EuiIconTip
+            type="questionInCircle"
+            content={
+              <span>
+                <FormattedMessage
+                  id="xpack.snapshotRestore.featureStatesFormField.formRowLabelTooltip"
+                  defaultMessage="A feature state contains the indices, system indices and data streams used to store configurations, history, and other data for an Elastic feature."
+                />
+              </span>
+            }
+            iconProps={{
+              className: 'eui-alignTop',
+            }}
+          />
+        </>
+      }
+      labelAppend={
+        <EuiSwitch
+          compressed
+          label={
+            <>
+              <FormattedMessage
+                id="xpack.snapshotRestore.featureStatesFormField.includeNoneLabel"
+                defaultMessage="Include none"
+              />{' '}
+              <EuiIconTip
+                type="questionInCircle"
+                content={
+                  <span>
+                    <FormattedMessage
+                      id="xpack.snapshotRestore.featureStatesFormField.includeNoneDescription"
+                      defaultMessage="All system indices can be omitted by including none of the feature states."
```

Comment:
This is potentially confusing--in the API, you have to explicitly specify none, an empty array includes all features. It might be good to make people explicitly select No features, rather than just deselecting each feature state.

Reply:
Ohh so basically the way I thought this could work is that if they:
* select nothing: all features are included
* select some: **only** the selected features are included in the snapshot
* toggle none: **no features** are included in the snapshot
![Screenshot 2022-05-06 at 15 12 56](https://user-images.githubusercontent.com/1191206/167138468-3f084b73-8337-458c-a7fa-e0da47f3f985.png)
![Screenshot 2022-05-06 at 15 12 51](https://user-images.githubusercontent.com/1191206/167138477-91775144-ee4d-422d-9d68-989904f6898a.png)
![Screenshot 2022-05-06 at 15 12 46](https://user-images.githubusercontent.com/1191206/167138480-471cad75-59d1-4334-accf-2b92c917b8c1.png)

Reply:
Would something like this work?


<img width="1079" alt="Screenshot 2022-05-06 at 17 24 56" src="https://user-images.githubusercontent.com/44808440/167163646-3176bf0d-8e24-43bc-9d49-4f67a03fd135.png">

Reply:
Depending on how much or how little we want to highlight it, we can go with text grouped buttons (which are grey)